linode/manager
# refactor: [M3-6875, M3-6719] - Upgrade remaining `LinodeSelect`s

## Description 📝
Finishes migrating instances of `LinodeSelect` over to the V2 component introduced in #9193. Also fixes a bug causing the input field of `LinodeSelectV2` to reset while the user is typing.

## Major Changes 🔄
- Remove old `LinodeSelect` with associated files
- Allow for IP address to replace option label using `showIPAddressLabel`
- Filter linodes by region and options filter
- Move `✓` to the right and remove `x` for selected options. This is because there's always multiple intuitive ways to remove a selected option, and to allow for flags on the left.

## Preview 📷

| Before  | After   |
| ------- | ------- |
| <img width="575" alt="Screenshot 2023-07-19 at 10 37 44 AM" src="https://github.com/linode/manager/assets/125309814/9cab44f8-9a83-48dc-bce9-81430c9b19b7"> | <img width="581" alt="Screenshot 2023-07-19 at 10 37 22 AM" src="https://github.com/linode/manager/assets/125309814/a99e2f31-f93f-4fe8-b314-300cbedc3fee"> |
| <img width="365" alt="Screenshot 2023-07-19 at 10 38 59 AM" src="https://github.com/linode/manager/assets/125309814/61cfbca8-f9bd-4892-85ac-94865746ecf4"> | <img width="366" alt="Screenshot 2023-07-19 at 10 47 40 AM" src="https://github.com/linode/manager/assets/125309814/dd3dc67d-e894-44ab-aa80-9f8e438247f6"> |
| <img width="366" alt="Screenshot 2023-07-19 at 10 47 59 AM" src="https://github.com/linode/manager/assets/125309814/9d4bcffc-68f5-474a-b7b7-2e97071b07d8"> | <img width="368" alt="Screenshot 2023-07-19 at 10 47 48 AM" src="https://github.com/linode/manager/assets/125309814/a26794fa-d4da-41cd-abdf-d7c5b6ff6645"> |
| <img width="428" alt="Screenshot 2023-07-19 at 10 58 41 AM" src="https://github.com/linode/manager/assets/125309814/305576fa-f70c-44dd-b6a7-72fd48bccd01"> | <img width="436" alt="Screenshot 2023-07-19 at 10 58 27 AM" src="https://github.com/linode/manager/assets/125309814/ab82a836-6a58-4415-bc5a-881f1ac8088c"> |

## How to test 🧪
1. Checkout branch or view preview link
2. Check out areas we use LinodeSelectV2
    - http://localhost:3000/firewalls (create)
    - http://localhost:3000/volumes/create
    - http://localhost:3000/nodebalancers/create
4. Ensure that when a value/option is selected, you can still view other options within region
5. Ensure when a region has changed, linode select has been cleared
